### PR TITLE
Username text-search endpoint

### PR DESCRIPTION
**Pull request by andysbolton**
This should include an endpoint for doing a text-search on usernames. The implementation of this will likely change in the future but right now is using the full-text search method from [here](http://rachbelaid.com/postgres-full-text-search-is-good-enough/).

### CODE REVIEW

File: database/database_test.go [hunk 1]
```diff
@@ -451,6 +451,36 @@ func TestGetUser(t *testing.T) {
...
+	if err != nil {
+		t.Fatalf("An error %s occurred when opening stub database connection", err)
+	}
+	defer DB.Close()
+
+	row := sqlmock.NewRows([]string{"id", "username"}).
+		AddRow("1", "CoolGuy420").
+		AddRow("2", "CoolGuyChiller")
+
+	mock.ExpectQuery(`SELECT (.+) FROM board.user`).WillReturnRows(row)
```
Comment (DarthHater) [inline database/database_test.go]: The test still works? This pattern seems like it would fail.

File: database/database.go [hunk 2]
```diff
@@ -140,6 +142,36 @@ func (d *Database) GetUser(username string) (user model.User, err error) {
 	return user, nil
 }
 
+// GetUsers will return a list of users whose username matches a search string.
+func (d *Database) GetUsers(search string) ([]model.User, error) {
+	users := []model.User{}
+	sqlStatement := `
+		SELECT Id, Username
+		FROM (SELECT Id, to_tsvector(Username) as lex, UserRole, Username
```
Comment (DarthHater) [inline database/database.go]: Really cool that you used this! Does it do fuzzy stuff?

File: main.go [hunk 3]
```diff
@@ -353,6 +353,7 @@ func setupRouter(d database.IDatabase) *gin.Engine {
 
 		auth.GET("/message/:messageid", func(c *gin.Context) {
 			messageID := c.Param("messageid")
+			fmt.Println("sup")
```
Comment (DarthHater) [inline main.go]: LOL might wanna remove this

File: database/database.go (hunk 2 above)
Comment (andysbolton) [reply database/database.go]: I don't think this one does fuzzy stuff, but just noticed that article has some info on fuzzy search, I'll take a look at that.

File: database/database.go [hunk 4]
```diff
@@ -140,6 +142,36 @@ func (d *Database) GetUser(username string) (user model.User, err error) {
...
+				FROM board.user) doc
+		WHERE doc.lex @@ to_tsquery($1) AND UserRole != 2`
```
Comment (DarthHater) [inline database/database.go]: I actually dug it here where you had the UserRole not hard coded, any reason to switch it to having it hard coded?

**Pull request merged**